lib/publiccloud/azure.pm

```diff
@@ -124,16 +131,61 @@ sub upload_img {
 
     my $key = $self->get_storage_account_keys($storage_account);
 
-    assert_script_run('az storage blob upload --max-connections 4 --account-name '
-          . $storage_account . ' --account-key ' . $key . ' --container-name ' . $self->container
-          . ' --type page --file ' . $file . ' --name ' . $img_name, timeout => 60 * 60 * 2);
-    assert_script_run('az disk create --resource-group ' . $self->resource_group . ' --name ' . $disk_name
-          . ' --source https://' . $storage_account . '.blob.core.windows.net/' . $self->container . '/' . $img_name
-          . ' --hyper-v-generation=' . $sku . ' --architecture=' . $arch);
 
-    assert_script_run('az image create --resource-group ' . $self->resource_group . ' --name ' . $img_name
-          . ' --os-type Linux --hyper-v-generation=' . $sku . ' --source=' . $disk_name);
 
+    # Check if blob already exists
+    my $container = $self->container;
+    my $blobs = script_output("az storage blob list --container-name '$container' --account-name '$storage_account' | jq '.[].name'");
+    $blobs =~ s/^\s+|\s+$//g;    # trim
+    record_info("blobs", $blobs);
+    my @blobs = split(/\n/, $blobs);
+    if (grep(/$img_name/, @blobs)) {
+        record_info('blob', "Blob already exists, omitting upload");
+    } else {
+        # Upload blob to Azure. Keep in mind that VM images need to be a page blob type
+        assert_script_run('az storage blob upload --max-connections 4 --account-name '
+              . $storage_account . ' --account-key ' . $key . ' --container-name ' . $self->container
+              . ' --type page --file ' . $file . ' --name ' . $img_name, timeout => 60 * 60 * 2);
+    }
+
+    if ($arch eq 'Arm64') {
+        # For Arm64 images we need to use the image galleries
+        my $publisher = get_var("PUBLIC_CLOUD_AZURE_PUBLISHER", "qe-c");
+        my $offer = get_required_var("PUBLIC_CLOUD_AZURE_OFFER");
+        my $definition = get_required_var('DISTRI') . '-' . get_required_var('FLAVOR') . '-' . get_required_var('VERSION');
+        $definition = get_var("PUBLIC_CLOUD_AZURE_IMAGE_DEFINITION", uc($definition));
+        $sku = get_required_var("PUBLIC_CLOUD_AZURE_SKU");
+        my $target_regions = get_var("PUBLIC_CLOUD_REGION", "westeurope");
```

Comment:
why plural and comma ? I don't remember that we need/support upload to several regions ?

Reply:
It's not yet needed, but it is supported by the tooling. And I don't see any downsides in allowing this from the side of `upload_img`, or do you?

Reply:
Also, if we ever want to test Arm64 images on other regions as well, we would need to extend the list here. This code allows this already, so all we would need to do, is to change the variables.

Reply:
if you will put several regions into that variable I assume you will get several failures in another places . So it may be supported by **Azure** but I have doubts about our test code . Please provide VR which proves me wrong or revert

Reply:
Here is a passing test run with `PUBLIC_CLOUD_REGION=westeurope,northeurope,westus,eastus2`: https://duck-norris.qe.suse.de/tests/11633#step/upload_image/70

And here is a screenshot from the Azure portal, showing the image version in the defined target regions

![image](https://user-images.githubusercontent.com/1527829/210599299-c114330d-59e8-43da-98a1-84aa61ea8d0a.png)

As far as I can see this works nicely.

Reply:
https://duck-norris.qe.suse.de/tests/11633#step/upload_image/65 but upload didn't happened with this job

Reply:
resolved